modules/Blockchain/Ethereum/index.js

```diff
@@ -1400,6 +1545,26 @@ class Ethereum {
         });
         return totalAmount.toString();
     }
+
+    async _getAverageGasPrice() {
```

Comment:
- move to a separate service, where the gasPrice value will be determined when needed (so that not every call to a blockchain function invokes an API call to external services) - i.e. store the latest determined value in the config and use it if it is not older than 30mins (if it is, get the latest price again)
- use web3.eth.getGasPrice and EthGasStation together to form the price, so that not only one service is consulted
- take the larger from them and add a safety margin (20%)
- if both return invalid values (non-reachable), use a previously fetched value if existent, otherwise use a default value from config